rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -154,6 +182,128 @@ pub fn gunzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, F
         }
     }
 }
+/// Takes seven named arguments sec, min, hour, mday, mon, year, isdst and returns the Unix time.
+pub fn mktime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+
+    let sec = match register.named("sec") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let min = match register.named("min") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let hour = match register.named("hour") {
+        Some(ContextType::Value(NaslValue::Number(x))) =>  *x as u32,
+        _ => 0,
+    };
+    let mday = match register.named("mday") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let mon = match register.named("mon") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 1,
+    };
+    let year = match register.named("year") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => 0,
+    };
+
+    // TODO: fix isdst
+    let isdst = match register.named("isdst") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => -1,
+    };
+
+    let offset = chrono::Local::now().offset().fix().local_minus_utc();
+    let r_dt = Utc.with_ymd_and_hms(year, mon, mday, hour, min, sec);
+    match r_dt {
+        LocalResult::Single(x) => Ok(NaslValue::Number(x.naive_local().timestamp() - offset as i64)),
+        _ => Ok(NaslValue::Null),
+    }
+}
+
+
+
+/// Returns an dict(mday, mon, min, wday, sec, yday, isdst, year, hour) based on optional given time in seconds and optional flag if utc or not.
+pub fn localtime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let named = register.named("utc");
+    let utc_flag;
```

Comment:
unnecessary define and later assignment, use match to return and assign immediately.